### DISCUSSION TITLE
Sending embed with interaction.reply works but not with channel.send (permissions error)

### DISCUSSION THREAD

**Discussion by ghost** [discordjs/discord.js#7545] [Q&A] [answered]
### Which package is this bug report for?

discord.js

### Issue description

https://github.com/kennedybaird/discord-embed-error

Clone repository, npm install, create bot and add to testing server.

Use command `/list-item issue: 2` ephemeral message will populate with embed

Use command `/list-item issue: 1` tries to use channel.send to send an embed, but gives permissions error.

Error does not appear when setting bot permissions to administrator.

Permissions as so:

![image](https://user-images.githubusercontent.com/20593811/155623592-39f5968d-b846-4f5e-a3c6-f1d121cafc8d.png)

### Package version

v13

### Node.js version

16.13.2

### Operating system

Linux

### Priority this issue should have

Medium (should be fixed soon)

### Which partials do you have configured?

No Partials

### Which gateway intents are you subscribing to?

Guilds, GuildMessages

**Comment by Jiralite** [comment]
This isn't to do with discord.js, it's just as you're told: you don't have permissions. Additionally, the screenshot you provided doesn't have embed links enabled and people may take permissions away from your bot.

**Comment by ghost** [comment]
Thanks for the reply.

I am starting to wonder if it is an issue with discordjs as I am able to edit the reply with an embed, without those permissions.

Even with those permissions I cannot send the embed successfully.

Why can I edit the reply with an embed, but not send a new message to the channel with the embed?

**Comment by monbrey** [comment]
Permissions errors come from the Discord API, not discord.js. The library does not perform internal permissions checks.

**Comment by ghost** [comment]
Right, but that's still not explaining why it works when it's via `interaction.reply` and not working when it's from `channel.send`?

**Reply by MCilento93** [reply]
Same issue here

**Comment by monbrey** [comment]
Interaction replies use Webhooks to respond - their permissions are inherently tied to the everyone role, not your bot user.

**Comment by ghost** [comment]
Ah, that makes sense.

I have tried sitting there putting on all types of permissions for the deployment, and I can't seem to get it work.

It doesn't work with embed links turned on either as the user above suggested. Do you have any immediate thoughts?

I really appreciate the help - i've been stuck on this for ages

**Comment by ghost** [comment]
For example - putting every permission except for Administrator it still doesn't work:

![image](https://user-images.githubusercontent.com/20593811/155625743-128f375b-bbba-456a-ab58-3b94a6cc35e0.png)

Is it something I'm missing with Intents?

```js
const client = new Client({
  intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES],
});```

**Comment by SpaceEEC** [comment]
Changing the values for the invite link won't change any already set permissions.
You need to go into the relevant guild and grant the bot the permission(s) there.

**Comment by ghost** [comment]
> Changing the values for the invite link won't change any already set permissions. You need to go into the relevant guild and grant the bot the permission(s) there.

Changing them and re-adding to the server does update the role assigned to the bot.

**Comment by SpaceEEC** [comment, marked as answer]
If the bot is having those permission on its integration role and you are getting that error when sending a message, then there is an overwrite denying the permission(s) in place.

**Comment by ghost** [comment]
Thanks - where can I troubleshoot what may be overwriting?

**Comment by ghost** [comment]
@SpaceEEC  Ok, that was perfect, I found that `@everyone` had embed-links turned off for the channel.

Is there a way to have my bot not be defined / affected by over-writes like that?

**Comment by SpaceEEC** [comment]
Only by granting it the administrator permission. (You could also create an overwrite granting for the bot / integration role, allowing that bit, behaves slightly different, but would also work here)

**Reply by ghost** [reply]
Thanks

**Comment by ghost** [comment]
Can close this (p.s: I did reach out in discord well before posting here)
